docker/deploy/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Multistage build: here we import the current source code
+# into build environment image, build the project, bundle it
+# and then extract it into a small image that just contains
+# the binaries we need to run
+
+ARG BUILD_ENV
+
+FROM $BUILD_ENV as builder
+
+# Build arbor
+COPY . /arbor
+
+# Build and bundle binaries
+RUN mkdir /arbor/build && cd /arbor/build && \
+    CC=mpicc CXX=mpicxx cmake .. \
+      -DARB_VECTORIZE=ON \
+      -DARB_ARCH=broadwell \
+      -DARB_WITH_PYTHON=OFF \
+      -DARB_WITH_MPI=ON \
+      -DARB_WITH_GPU=ON \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DCMAKE_INSTALL_PREFIX=/usr && \
+    make -j$(nproc) tests && \
+    /root/bundler/bundler \
+      -d /root/arbor.bundle \
+      -e /arbor/build/bin/modcc \
+      -e /arbor/build/bin/unit \
+      -e /arbor/build/bin/unit-local \
+      -e /arbor/build/bin/unit-modcc \
+      -e /arbor/build/bin/unit-mpi && \
+    rm -rf /arbor
+
+FROM ubuntu:18.04
+
+# This is the only thing necessary really from nvidia/cuda's ubuntu18.04 runtime image
+ENV NVIDIA_VISIBLE_DEVICES all
+ENV NVIDIA_DRIVER_CAPABILITIES compute,utility
+ENV NVIDIA_REQUIRE_CUDA "cuda>=10.1 brand=tesla,driver>=384,driver<385 brand=tesla,driver>=396,driver<397 brand=tesla,driver>=410,driver<411"
```

Comment:
Why is `NVIDIA_REQUIRE_CUDA` needed?

Reply:
Normally you would extend NVIDIA's base image, but that's quite big (1GB) as it pulls in the runtime, cublas etc. The way docker and sarus support gpu's is via NVIDIA Container Toolkit, which as a bare minimum just requires those environment variables in the container which are used to check whether the driver (outside of the container) and the container are compatible.

See https://gitlab.com/nvidia/container-images/cuda/-/blob/master/dist/ubuntu18.04/10.1/base/Dockerfile#L30-33 for reference.